lib/pbench/server/indexer.py

```diff
@@ -1536,121 +1539,113 @@ def _make_source_json(self):
-                # sample result.json - we ignore each sample result.json since
-                # each sample is contained in the experiment level
-                # result.json.
-                continue
-            elif dirname != self.ptb.dirname:
-                # iteration result.json - there is no value in processing the
-                # result.json file at the iteration level because everything
-                # it contains is also in the experiment level's result.json.
-                continue
+        result_json = os.path.join(
+            self.ptb.extracted_root, self.ptb.dirname, "result.json"
+        )
+        try:
+            # Read the file and interpret it as a JSON document.
+            with open(result_json) as fp:
+                results = json.load(fp)
+        except Exception as e:
+            self.logger.warning(
+                "result-data-indexing: encountered invalid JSON file," " {}: {:r} ({})",
+                result_json,
+                e,
+                self.ptb._tbctx,
+            )
+            self.counters["not_valid_json_file"] += 1
+            return
 
-            result_json = os.path.join(self.ptb.extracted_root, dirname, "result.json")
-            try:
-                # Read the file and interpret it as a JSON document.
-                with open(result_json) as fp:
-                    results = json.load(fp)
-            except Exception as e:
-                self.logger.warning(
-                    "result-data-indexing: encountered invalid JSON file,"
-                    " {}: {:r} ({})",
-                    result_json,
-                    e,
-                    self.ptb._tbctx,
-                )
-                self.counters["not_valid_json_file"] += 1
-                continue
+        # The outer results object should be an array of iterations. Probe
+        # to see if that is true.
+        if not isinstance(results, list):
+            self.logger.warning(
+                "result-data-indexing: encountered unexpected"
+                " JSON file format, %s ({})",
+                result_json,
+                self.ptb._tbctx,
+            )
+            return
 
-            # The outer results object should be an array of iterations. Probe
-            # to see if that is true.
-            if not isinstance(results, list):
+        for iteration in results:
+            try:
+                iter_number = iteration["iteration_number"]
+                iter_name = iteration["iteration_name"]
+                iter_data = iteration["iteration_data"]
+            except KeyError:
                 self.logger.warning(
-                    "result-data-indexing: encountered unexpected"
-                    " JSON file format, %s ({})",
+                    "result-data-indexing: could not find"
+                    " iteration data in JSON file, {} ({})",
                     result_json,
                     self.ptb._tbctx,
```

Comment:
I shouldn't worry about this since it's unusual and weird, and trying to identify a specific broken iteration would be awkward... but somehow it makes me uncomfortable. So instead of offering specific criticism or correction, I'll simply make myself feel better by sharing that it makes me vaguely uncomfortable. ;-)